### DISCUSSION TITLE
mapping applications to prototypes

### DISCUSSION THREAD

**Discussion by pv2-0** [eclipse/mosaic#66] [Q&A] [answered]
If I create my own application and have the compiled java file in the "application" folder of my scenario, how should I call it from my mapping file?
In all of the examples, they have "org.eclipse.mosaic. ..." at the beginning. Would this be the same for applications that I create?

In my application, vehicles should be sending V2X messages, but in the visualizer, none of the vehicles ever turn green or red, so I assume that my application is not linking properly to the prototype.

Thanks!

**Comment by kschrab** [comment, marked as answer]
As stated [in the documentation](https://www.eclipse.org/mosaic/docs/develop_applications/#create-a-hello-world-application-based-on-maven), you need to use the full qualified name of the application, that is: `name.of.the.package.ClassName`. If your application is not inside a Java package, you can just use the name of the class. If your class could not be loaded, there would be an error and the simulation would not succeed.
